### ISSUE TITLE
Autofocus in video mode

### ISSUE THREAD

**Issue by vgevorgyan** [sandrios/sandriosCamera#7]
Hello,

I'm trying to record video with the following code:
```java
Intent cameraIntent;
if (CameraHelper.hasCamera2(this)) {
    cameraIntent = new Intent(this, Camera2Activity.class);
} else {
    cameraIntent = new Intent(this, Camera1Activity.class);
}
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.REQUEST_CODE, REQUEST_CODE_VIDEO);
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.SHOW_PICKER, false);
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.MEDIA_ACTION, SandriosCameraConfiguration.MEDIA_ACTION_VIDEO);
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.MEDIA_QUALITY, SandriosCameraConfiguration.MEDIA_QUALITY_HIGHEST);
startActivityForResult(cameraIntent, REQUEST_CODE_VIDEO);
```
but autofocus not working in preview and during recording.

But that is working when I'm trying to take a picture:

```java
Intent cameraIntent;
if (CameraHelper.hasCamera2(this)) {
    cameraIntent = new Intent(this, Camera2Activity.class);
} else {
    cameraIntent = new Intent(this, Camera1Activity.class);
}
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.REQUEST_CODE, REQUEST_CODE_PICTURE);
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.SHOW_PICKER, false);
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.MEDIA_ACTION, SandriosCameraConfiguration.MEDIA_ACTION_PHOTO);
cameraIntent.putExtra(SandriosCameraConfiguration.Arguments.MEDIA_QUALITY, SandriosCameraConfiguration.MEDIA_QUALITY_HIGHEST);
startActivityForResult(cameraIntent, REQUEST_CODE_PICTURE);
```

Should I put any configuration to allow camera autofocus in video mode?

Also here is my manifest:

```xml
<uses-feature android:glEsVersion="0x00020000" android:required="false" />
<uses-feature android:name="android.hardware.camera2.full" android:required="false" />
<uses-feature android:name="android.hardware.camera" android:required="false" />
<uses-feature android:name="android.hardware.camera.autofocus" android:required="false" />
<uses-feature android:name="android.hardware.telephony" android:required="false" />
```

Thanks

**Comment by rameshvoltella** [comment]
works fine in samsung device but fail in motorola xplay 
objects in distance is not focused but object closer is focused, how to fix this

**Comment by vgevorgyan** [comment]
In my case it's not working on all samsung devices, for example on Samsung Galaxy A7 and old Samsung Galaxy S4 not working, but on Samsung Galaxy S6 edge it's working.

**Comment by vgevorgyan** [comment]
Also, not working on Nexus 7.

**Comment by arpitgandhi9** [comment]
@vgevorgyan Hey, Tried working with auto focus on Nexus 7 on Android 4.3 since it was only device available with me having the issue. I am not able to see any issues with Auto Focus. Can you please provide some more information on this.. 

```
new SandriosCamera(activity, CAPTURE_MEDIA)
                            .setShowPicker(true)
                            .setVideoFileSize(20)
                            .setMediaAction(CameraConfiguration.MEDIA_ACTION_BOTH)
                            .enableImageCropping(true)
                            .launchCamera();
```

Pull requests to develop branch are welcome.

**Comment by asujiman** [comment]
I found that issue but not sure it's any effect?
This bug come from Camera1Manager class 
See this line

if (configurationProvider.getMediaAction() == CameraConfiguration.MEDIA_ACTION_PHOTO
                    || configurationProvider.getMediaAction() == CameraConfiguration.MEDIA_ACTION_BOTH)
                turnPhotoCameraFeaturesOn(camera, parameters);
            else if (configurationProvider.getMediaAction() == CameraConfiguration.MEDIA_ACTION_PHOTO){
                turnVideoCameraFeaturesOn(camera, parameters);
}

If we set 
setMediaAction(CameraConfiguration.MEDIA_ACTION_VIDEO) 

Camera will not auto focus coz didn't call turnVideoCameraFeaturesOn

So I changed it to 
if (configurationProvider.getMediaAction() == CameraConfiguration.MEDIA_ACTION_PHOTO
                    || configurationProvider.getMediaAction() == CameraConfiguration.MEDIA_ACTION_BOTH)
                turnPhotoCameraFeaturesOn(camera, parameters);
            else if (configurationProvider.getMediaAction() == CameraConfiguration.MEDIA_ACTION_VIDEO){
                turnVideoCameraFeaturesOn(camera, parameters);
}

After that auto focus is working fine

Hope this help sir.

**Comment by arpitgandhi9** [comment]
@asujiman Thank you for your contribution.
Changes should be available with 1.1.0.. Coming Soon..
